### DISCUSSION TITLE
The next client attempts to use a PageLoader to pull in the chunk for the file-named pathname at runtime.

### DISCUSSION THREAD

**Discussion by sharunthomas** [vercel/next.js#15862] [Help]
# Bug

## Created a custom server with express, entire page reloads when route to custom route path

Below is my server.js file, I created two pages a.js and b.js and routed with URL `/atest` and `/btest`
When I try to load the page a using URL `/atest ` using `Link`  Next client tries to load atest.js and gives error then the entire app re-renders which is not an acceptable behaviour.

After reloads it will give the necessary pages but entire page reload gives a bad look for the app

You can download this application from 
https://github.com/sharunthomas/custom-server-next-prototype

Or you can clone project from https://github.com/sharunthomas/custom-server-next-prototype.git
## To Reproduce
```/** Custom server using express server*/


const express = require('express')
const next = require('next')

const port = parseInt(process.env.PORT, 10) || 3000
const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev , conf:{useFileSystemPublicRoutes: false}})
const handle = app.getRequestHandler()

app.prepare().then(() => {
  const server = express()

  server.get('/atest', (req, res) => {
    return app.render(req, res, '/a', req.query)
  })

  server.get('/btest', (req, res) => {
    return app.render(req, res, '/b', req.query)
  })

  server.all('*', (req, res) => {
    return handle(req, res)
  })

  server.listen(port, (err) => {
    if (err) throw err
    console.log(`> Ready on http://localhost:${port}`)
  })
})

```

## System information

- OS:  Windows
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: 9.4.4

**Comment by will-stone** [comment]
Hi. Did you ever find a solution for this?
